mbuild/formats/gomc_conf_writer.py

```diff
@@ -4659,6 +5586,160 @@ def __init__(
             )
             raise ValueError(print_error_message)
 
+        # check that TargetedSwap_DataInput values are provided if self.TargetedSwapFreq
+        # and self.IntraTargetedSwapFreq is not zero
+        if self.TargetedSwap_DataInput is None and (
+            self.TargetedSwapFreq != 0 or self.IntraTargetedSwapFreq != 0
+        ):
+            self.input_error = True
+            print_error_message = (
+                "ERROR: The TargetedSwap_DataInput variable is equal to None, "
+                "but the TargetedSwapFreq or IntraTargetedSwapFreq move ratio is non-zero."
+            )
+            raise ValueError(print_error_message)
+
+        # check that either the ChemPot or Fugacity are set for the Target swap moves
+        # only ChemPot or Fugacity can be set
+        if isinstance(self.TargetedSwap_DataInput, dict):
+            chempot_used = False
+            fugacity_used = False
+            for tag_id_keys_j in list(self.TargetedSwap_DataInput.keys()):
+                target_swap_tag_id_dict_key_data = self.TargetedSwap_DataInput[
+                    tag_id_keys_j
+                ]
+
+                for target_swap_dict_key_j_lower in list(
+                    target_swap_tag_id_dict_key_data.keys()
+                ):
+                    if target_swap_dict_key_j_lower in ["subvolumechempot"]:
+                        chempot_used = True
+
+                    if target_swap_dict_key_j_lower in ["subvolumefugacity"]:
+                        fugacity_used = True
+
+            if chempot_used is True and fugacity_used is True:
+                print_error_message = (
+                    "Both ChemPot and Fugacity were used in the "
+                    "TargetedSwap_DataInput dictionaries. "
+                    "However, only ChemPot or Fugacity may be used, not both."
+                )
+                raise ValueError(print_error_message)
+
+            if (chempot_used is True and self.Fugacity is not None) or (
+                fugacity_used is True and self.ChemPot is not None
+            ):
+                print_error_message = (
+                    "Both ChemPot and Fugacity were used in the "
+                    "TargetedSwap_DataInput dictionaries "
+                    "and in the standard GOMC swap inputs. "
+                    "However, only ChemPot or Fugacity may be used, not both."
+                )
+                raise ValueError(print_error_message)
+
+            if (
+                chempot_used is True or fugacity_used is True
+            ) and self.ensemble_type in ["NPT", "NVT", "GEMC_NVT", "GEMC_NPT"]:
+                print_error_message = (
+                    "Either the ChemPot and Fugacity were used in the "
+                    "TargetedSwap_DataInput dictionaries, "
```

Comment:
Yeah, I agree this may be cleanup up a bit in the future.  For now however, since it is working, I would like to focus on getting MoSDeF-GMSO-GOMC working, before making structural changes to the existing code(s).